internal/moreos/proc_linux_test.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2021 Tetrate
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package moreos
+
+import (
+	"os"
+	"os/exec"
+	"os/signal"
+	"path/filepath"
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+// TestCmdStart tests CmdStart by starting a parent process that spawn a child process and then sends
+// signal to terminate the child process which in effect will terminate the parent process via
+// the configured Pdeathsig.
+func TestCmdStart(t *testing.T) {
+	tempDir := t.TempDir()
+
+	cmd := newCmdForTest("TestCmdStart_Parent", tempDir)
+	err := cmd.Run()
+	require.NoError(t, err)
+
+	require.FileExists(t, filepath.Join(tempDir, "parent-sigterm"))
+	require.FileExists(t, filepath.Join(tempDir, "child-sigterm"))
+
+	p, err := os.Stat(filepath.Join(tempDir, "parent-sigterm"))
+	require.NoError(t, err)
```

Comment:
what if you called `t.TempDir()` twice instead of manually making subdirs then checking them? a lot less code right?